temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivityWorker.java

```diff
@@ -449,6 +474,20 @@ public void handle(LocalActivityAttemptTask attemptTask) throws Exception {
       MDC.put(LoggerTag.WORKFLOW_ID, activityTask.getWorkflowExecution().getWorkflowId());
       MDC.put(LoggerTag.WORKFLOW_TYPE, activityTask.getWorkflowType().getName());
       MDC.put(LoggerTag.RUN_ID, activityTask.getWorkflowExecution().getRunId());
+
+      // At this point the permit should definitely be set
+      if (executionContext.getPermit() == null) {
+        throw new IllegalStateException("Permit is expected to be set at this point");
```

Comment:
Is there a reason this check can't just be in the `TrackingSlotSupplier`?  I think its' current behaviour to silently do nothing on `null` permits contributed to us not noticing this issue sooner.

Reply:
Seems like it could be in `markUsed`. I will verify and move it there.